app/controllers/category_controller.rb

```diff
@@ -36,8 +39,22 @@ def create
   def delete
     anchor = @category.parent_id.present? ? "#category-#{@category.parent_id}" : nil
     @category.destroy #_but_attach_children_to_parent
+    
     flash[:notice] = "Category has been deleted"
     redirect_to "#{request.env['HTTP_REFERER']}#{anchor}"
   end
-
+  def enable_gis
+    @category.update_attribute(:gis_enabled, true)
+    @category.descendants.each {|d| d.update_attribute(:gis_enabled, true)}
+    
+    flash[:notice] = "GIS Enabled for #{@category.title}"
+    ajax_redirect_to "#{request.env['HTTP_REFERER']}#category-#{@category.id}"
```

Comment:
Path generation by mucking about with `HTTP_REFERER` is pretty ugly.  Do we already have a category_path URL helper we can use?  If not, it would be worth making one.  (We might need to pass in the controller as well.)

Reply:
This whole controller seemed to use that method, so I went with it, assuming that there was something about it that I didn't know. My guess was that it had something to do with redirecting to specific anchors, though, that is perfectly doable with path helpers. I'll fix them all.